rules/DowngradePhp80/Rector/MethodCall/DowngradeNamedArgumentRector.php

```diff
@@ -96,124 +93,50 @@ public function refactor(Node $node): ?Node
-            $className = (string) $new->getAttribute(AttributeKey::CLASS_NAME);
-        }
-
-        if (! $this->reflectionProvider->hasClass($className)) {
-            return null;
-        }
-
-        $classReflection = $this->reflectionProvider->getClass($className);
-        $className = Strings::after($className, '\\', -1);
-        return $this->getCallerNodeFromClassReflection($new, $classReflection, $className);
-    }
-
-    /**
-     * @param MethodCall|StaticCall|New_ $node
-     */
-    private function getCaller(Node $node): ?Node
+    private function removeNamedArguments(MethodCall | StaticCall | New_ $node, array $args): ?Node
     {
         if ($node instanceof New_) {
-            return $this->getClassMethodOfNew($node);
-        }
-
-        $caller = $node instanceof StaticCall
-            ? $this->nodeRepository->findClassMethodByStaticCall($node)
-            : $this->nodeRepository->findClassMethodByMethodCall($node);
-
-        if ($caller instanceof ClassMethod) {
-            return $caller;
-        }
-
-        $type = $this->callTypeAnalyzer->resolveCallerType($node);
-        if (! $type instanceof ObjectType) {
-            return null;
-        }
-
-        $classReflection = $this->reflectionProvider->getClass($type->getClassName());
-        return $this->getCallerNodeFromClassReflection($node, $classReflection);
-    }
-
-    /**
-     * @param MethodCall|StaticCall|New_ $node
-     */
-    private function getCallerNodeFromClassReflection(
-        Node $node,
-        ClassReflection $classReflection,
-        ?string $className = null
-    ): ?Node {
-        $fileName = $classReflection->getFileName();
-        if (! is_string($fileName)) {
-            return null;
-        }
-
-        $stmts = $this->parser->parse($this->smartFileSystem->readFile($fileName));
-        if ($node instanceof New_) {
-            /** @var string $className */
-            $class = $this->betterNodeFinder->findFirst((array) $stmts, function (Node $node) use ($className): bool {
-                if (! $node instanceof Class_) {
-                    return false;
-                }
-
-                return $this->isName($node, $className);
-            });
-
-            if (! $class instanceof Class_) {
+            $methodReflection = $this->callReflectionResolver->resolveConstructor($node);
+            if ($methodReflection === null) {
                 return null;
             }
 
-            return $class->getMethod(MethodName::CONSTRUCT);
+            return $this->processRemoveNamedArgument($methodReflection, $node, $args);
         }
 
-        return $this->betterNodeFinder->findFirst((array) $stmts, function (Node $n) use ($node): bool {
-            if (! $n instanceof ClassMethod) {
-                return false;
-            }
+        $callerReflection = $this->callReflectionResolver->resolveCall($node);
+        if ($callerReflection === null) {
+            return null;
+        }
 
```

Comment:
@samsonasik Just an update on Rector achitecture. We're moving toward reflection for anything that is *read-only*. That way we can use parallel run in the future and let PHPStan static reflection handle any metadata we'll need.

I've added reflection resolver here, it will save us so much manual work around nodes.